contracts/cw4-group-sorted/src/msg.rs

```diff
@@ -0,0 +1,57 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::query::Member;
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct InstantiateMsg {
+    /// The admin has the ability to add new items to the member list
+    /// and to update the admin. Omit the admin to make the member
+    /// list immutable.
+    pub admin: Option<String>,
+    /// The members to initialize the contract with.
+    pub members: Vec<Member>,
+}
+
+#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
+#[serde(rename_all = "snake_case")]
+pub enum ExecuteMsg {
+    /// Change the admin
+    UpdateAdmin { admin: Option<String> },
+    /// Applys a diff to the existing members. Remove is applied
+    /// after add, so if an address is in both, it is removed
+    UpdateMembers {
+        remove: Vec<String>,
+        add: Vec<Member>,
```

Comment:
are there any benefits of coupling these two? (suppose in the future we have a call to this execute method for, it would be weird to pass an empty set to the add and only add to the remove).

Reply:
We're trying to be compatible with the [cw4 spec](https://github.com/CosmWasm/cw-plus/blob/main/packages/cw4/README.md) which requires that these are implemented. Think an advantage of this approach is that gas fees have a high base cost and then scale more slowly with an increase in required compute. Doing 'more' in one operation is liable to be more gas efficient than two smaller operations.